`azurerm_eventhub` - add `storage_authentication`, `storage_authentication_id` in capture description feature (#31265)
terraform-providers/terraform-provider-azurerm: website/docs/r/eventhub.html.markdown

* `storage_account_id` - (Required) The ID of the Blob Storage Account where messages should be archived.

* `storage_authentication_type` - (Optional) The identity type used to authenticate the Blob Storage Account where messages should be archived. Possible values are `SystemAssigned`, `UserAssigned`.

> * `storage_authentication_id` - (Optional) The id of the user assigned identity that's used to authenticate the Blob Storage Account where messages should be archived. 

Reviewer: Technical documentation should not us contractions:
```suggestion
* `storage_authentication_id` - (Optional) The id of the user assigned identity that is used to authenticate the Blob Storage Account where messages should be archived. 
```

> * `storage_authentication_id` - (Optional) The id of the user assigned identity that is used to authenticate the Blob Storage Account where messages should be archived. 

Reviewer 2: ```suggestion
* `storage_authentication_id` - (Optional) The id of the User Assigned Identity that is used to authenticate the Blob Storage Account where messages should be archived. 
```

> ~> **Note:** The `system-assigned` or `user-assigned` managed identity must be `enabled` on the parent eventhub namespace, else the capture feature cannot be configured.

Reviewer 2: should this be `SystemAssigned` or `UserAssigned` ?

```suggestion
~> **Note:** The `SystemAssigned` or `UserAssigned` managed identity must be `enabled` on the parent eventhub namespace, in order for the capture feature to be configured.
```
